passwordComparator.js

```diff
@@ -0,0 +1,78 @@
+function comparePasswords() {
+    var password1 = document.getElementById("psw");
+    var length = document.getElementById("length");
+    var bigLetter = document.getElementById("bigLetter");
+    var smallLetter = document.getElementById("smallLetter");
+    var sign = document.getElementById("sign");
+
+    var passwordToCompare = document.getElementById("psw2");
+    var orDifferent = document.getElementById("comparison");
+
+// Gdy Input jest aktywny, pojawia się tabela wiadomości z brakującym znakiem
+    password1.onfocus = function() {
+        document.getElementById("message").style.display = "table-row";
+    }
+
+// Porównywanie po wpisaniu każdego kolejnego znaku
+    password1.onkeyup = function() {
+
+        // Sprawdź, czy zgadza się długość
+        if(password1.value.length >= 8) {
+            length.classList.remove("invalid");
+            length.classList.add("checked");
+        } else {
+            length.classList.remove("checked");
+            length.classList.add("invalid");
+        }
```

Comment:
Te ify się w znacznym stopniu powtarzają. Możesz to wydzielić do osobnej funkcji i wtedy znacznie skrócić ten kod.
Nie widzę też, żebyś miała gdziekolwiek zdefiniowaną klasę invalid, więc można ją usunąć zarówno z kodu JS jak i HTMLa